### ISSUE TITLE
ada is slower than linear？

### ISSUE THREAD

**Issue by nlper27149** [rosinality/adaptive-softmax-pytorch#2]
In model.py file , i add time module between 'self.linear' . As the result , ada got 5x slower than linear
`
linear = self.linear(output.contiguous() \
                .view(output.size(0) * output.size(1), output.size(2)))
`
change to
`
t1 = time.time()
linear = self.linear(output.contiguous() \
                .view(output.size(0) * output.size(1), output.size(2)))
t2 = time.time()
return linear, hidden, t2-t1
`

time-cost ada : 0.0005
time-cost linear :0.0001

**Comment by rosinality** [comment]
Efficiency is depend on token distributions, vocab sizes, and cutoff splits. And you will need torch.cuda.synchronize() for accurate timing.

**Comment by nlper27149** [comment]
Hi , I didn't change any params( token distributions, vocab sizes, and cutoff splits etc.) from your setting.  The result is same. I noticed that adaptive got speeding up because of back propagation (calculate loss). linear cost more time than ada on loss step.

**Comment by rosinality** [comment]
Yes, efficiency of adaptive softmax came from the combination with adaptive loss.
